packages/flutter/lib/src/rendering/sliver_list.dart

```diff
@@ -178,7 +178,7 @@ class RenderSliverList extends RenderSliverMultiBoxAdaptor {
         assert(earliestUsefulChild != null);
         final double firstChildScrollOffset = earliestScrollOffset - paintExtentOf(firstChild);
         geometry = SliverGeometry(
-          scrollOffsetCorrection: -firstChildScrollOffset,
+          scrollOffsetCorrection: firstChildScrollOffset == 0 ? null : -firstChildScrollOffset,
```

Comment:
We should do a while loop on `indexOf(firstChild) > 0)` and only return on line 185 if firstChildScrollOffset < -precisionErrorTolerance, other wise we should continue downward to finish the layout.

If we set it to null here and return on line 185, the rendering binding will think the layout is finished and proceed to painting.

Reply:
For example this may cause a problem if the children after the firstChild change their size in this perform layout.

Reply:
Ah, I think I see what you mean now. I'll update. Thanks!